### ISSUE TITLE
[1.16.5] Crash with Mana & Artifice Fire Damage

### ISSUE THREAD

**Issue by FoxMcloud5655** [dainxt/DungeonsMod-Issue-Tracker#43]
When someone casts a spell that causes a fire block to appear, Dungeons Mod encounters a NullPointerException.  Not sure which side is at fault here, but this is at the highest up in the stacktrace.

EDIT: Update; the player who caused the server to crash stated that they shot a Ghast fireball out of the air with their own fireball, so this explains why something like a Block at a BlockPos could be null or whatever this code is doing at the specified line of code.  If it was open-source, I could likely help fix it, but I trust you've got this :)

```
Caused by: java.lang.NullPointerException
at com.dainxt.dungeonsmod.handlers.EventsHandler.onEntityPlace(EventsHandler.java:226) ~[dungeonsmod:1.16.3-1.4.41] {re:classloading}
at net.minecraftforge.eventbus.ASMEventHandler_495_EventsHandler_onEntityPlace_EntityPlaceEvent.invoke(.dynamic) ~[?:?] {}
at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?] {}
at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?] {}
at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?] {}
at net.minecraftforge.event.ForgeEventFactory.onBlockPlace(ForgeEventFactory.java:168) ~[forge:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:runtimedistcleaner:A}
at com.ma.tools.BlockUtils.placeBlock(BlockUtils.java:123) ~[mana-and-artifice:1.5.0.10] {re:classloading}
at com.ma.spells.components.ComponentFireDamage.ApplyEffect(ComponentFireDamage.java:57) ~[mana-and-artifice:1.5.0.10] {re:classloading}
at com.ma.spells.SpellCaster.lambda$ApplyComponents$4(SpellCaster.java:298) ~[mana-and-artifice:1.5.0.10] {re:classloading}
at com.ma.spells.crafting.SpellRecipe.iterateComponents(SpellRecipe.java:189) ~[mana-and-artifice:1.5.0.10] {re:classloading}
at com.ma.spells.SpellCaster.ApplyComponents(SpellCaster.java:293) ~[mana-and-artifice:1.5.0.10] {re:classloading}
at com.ma.entities.sorcery.targeting.EntitySpellProjectile.applyEffect(EntitySpellProjectile.java:280) ~[mana-and-artifice:1.5.0.10] {re:classloading,pl:runtimedistcleaner:A}
at com.ma.entities.sorcery.targeting.EntitySpellProjectile.blockHit(EntitySpellProjectile.java:223) ~[mana-and-artifice:1.5.0.10] {re:classloading,pl:runtimedistcleaner:A}
at com.ma.entities.sorcery.targeting.EntitySpellProjectile.func_70227_a(EntitySpellProjectile.java:315) ~[mana-and-artifice:1.5.0.10] {re:classloading,pl:runtimedistcleaner:A}
at com.ma.entities.sorcery.targeting.EntitySpellProjectile.func_70071_h_(EntitySpellProjectile.java:459) ~[mana-and-artifice:1.5.0.10] {re:classloading,pl:runtimedistcleaner:A}
```

**Comment by dainxt** [comment]
Probably fixed on 1.4.42, Blocks and BlockPos should not be null in this event, but the entity can (Imagine a block placed by a null entity)... Let me know if the problem is fixed when possible.

**Comment by FoxMcloud5655** [comment]
I apologize for not getting back to you on this...  I had forgotten and let this fall by the wayside.

I've not seen the crash happen since updating, so this is likely fixed.
